freqtrade/plot/plotting.py

```diff
@@ -339,31 +342,34 @@ def analyse_and_plot_pairs(config: Dict[str, Any]):
 
     plot_elements = init_plotscript(config)
     trades = plot_elements['trades']
+    if trades is None:
+        raise ValueError('No trades to analyze')
+    else:
+        pair_counter = 0
+        for pair, data in plot_elements["tickers"].items():
+            pair_counter += 1
+            logger.info("analyse pair %s", pair)
+            tickers = {}
+            tickers[pair] = data
+
+            dataframe = strategy.analyze_ticker(tickers[pair], {'pair': pair})
+            logger.debug(f'pair: {pair}')
+
+            trades_pair = trades.loc[trades['pair'] == pair]
+            trades_pair = extract_trades_of_period(dataframe, trades_pair)
+
+            fig = generate_candlestick_graph(
+                pair=pair,
+                data=dataframe,
+                trades=trades_pair,
+                indicators1=config["indicators1"],
+                indicators2=config["indicators2"],
+            )
 
-    pair_counter = 0
-    for pair, data in plot_elements["tickers"].items():
-        pair_counter += 1
-        logger.info("analyse pair %s", pair)
-        tickers = {}
-        tickers[pair] = data
-
-        dataframe = strategy.analyze_ticker(tickers[pair], {'pair': pair})
-
-        trades_pair = trades.loc[trades['pair'] == pair]
-        trades_pair = extract_trades_of_period(dataframe, trades_pair)
-
-        fig = generate_candlestick_graph(
-            pair=pair,
-            data=dataframe,
-            trades=trades_pair,
-            indicators1=config["indicators1"],
-            indicators2=config["indicators2"],
-        )
-
-        store_plot_file(fig, filename=generate_plot_filename(pair, config['ticker_interval']),
-                        directory=config['user_data_dir'] / "plot")
+            store_plot_file(fig, filename=generate_plot_filename(pair, config['ticker_interval']),
+                            directory=config['user_data_dir'] / "plot")
 
-    logger.info('End of plotting process. %s plots generated', pair_counter)
+            logger.info('End of plotting process. %s plots generated', pair_counter)
```

Comment:
i think this is one intend too deep - it should be after the for loop, not within.

Reply:
removed the error check i the function and fixed indenting.